File: December2022/Java/DailyTemperatures.java
// @saorav21994

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {

        int l = temperatures.length;
        int [] res = new int[l];
        Stack<Integer> st = new Stack<>();

        for (int i = 0; i < l; i++) {
            if (st.isEmpty()) {
                st.push(i);
            }
            else {
                while (!st.isEmpty() && temperatures[i] > temperatures[st.peek()]) {
                    int top = st.pop();
                    res[top] = i-top;
                }
                st.push(i);
            }
        }

        return res;

    }
}

File: December2022/C++/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        stack<pair<int,int>>temp;
        
  
        int len = temperatures.size();
        
        vector<int>ans(len,0);
        
        
        for(int index = len-1; index>=0; index--){
            
            int currTemp = temperatures[index];
            
            while(!temp.empty() && temp.top().first<= currTemp){
                
                temp.pop();
                
            }
            
            if(temp.size()){
                
                ans[index] = temp.top().second - index;
            }
            
            temp.push({currTemp, index});
        }
        
        return ans;
    }
};